lib/bloc/account/account_bloc.dart

```diff
@@ -231,6 +238,68 @@ class AccountBloc {
     action.resolve(await _breezLib.sendWalletCoins(action.destAddress, Int64(action.feeRate)));     
   }
 
+  Future _exportPaymentsAction(ExportPayments action) async {
+    action.resolve(await _exportPayments());
+  }
+
+  Future _exportPayments() async {
+    // convert payment list to csv format and save to disk
+    return await _saveCsvFile(const ListToCsvConverter().convert(_generatePaymentList()));
+  }
+
+  _generatePaymentList(){
+    List<List<dynamic>> paymentList = new List.generate(_filterPayments(_paymentsController.value.paymentsList).length, (index) {
+      List paymentItem = new List();
+      paymentItem.add(DateUtils.formatYearMonthDayHourMinute(DateTime.fromMillisecondsSinceEpoch(
+          _filterPayments(_paymentsController.value.paymentsList).elementAt(index).creationTimestamp.toInt() * 1000)));
+      paymentItem.add(_filterPayments(_paymentsController.value.paymentsList).elementAt(index).title);
+      paymentItem.add(_filterPayments(_paymentsController.value.paymentsList).elementAt(index).description);
+      paymentItem.add(_filterPayments(_paymentsController.value.paymentsList).elementAt(index).destination);
+      paymentItem.add(_filterPayments(_paymentsController.value.paymentsList).elementAt(index).amount.toString());
+      paymentItem.add(_filterPayments(_paymentsController.value.paymentsList).elementAt(index).preimage);
+      paymentItem.add(_filterPayments(_paymentsController.value.paymentsList).elementAt(index).paymentHash);
+      return paymentItem;
+    });
+    paymentList.insert(0, ["Date & Time", "Title", "Description", "Node ID", "Amount", "Preimage", "TX Hash"]);
+    return paymentList;
+  }
+
+  Future _saveCsvFile(String csv) async {
+    try {
+      String filePath = await _createCsvFilePath();
+      final file = File(filePath);
+      await file.writeAsString(csv);
+      return file.path;
+    } catch (e) {
+      return e;
+    }
+  }
+
+  Future<String> _createCsvFilePath() async {
+    final directory = await getApplicationDocumentsDirectory();
+    String filePath = '${directory.path}/breez_payment_list';
+    filePath = appendFilterInformation(filePath);
+    filePath += ".csv";
+    return filePath;
+  }
+
+  String appendFilterInformation(String filePath) {
+    if (listEquals(_paymentFilterController.value.paymentType, [PaymentType.SENT, PaymentType.WITHDRAWAL])) {
+      String typeFilter = "sent";
```

Comment:
`typeFilter` variable is not used anywhere, it looks that appending either "sent" or "received" directly is simpler.

Reply:
[Removed typeFilter variable](https://github.com/breez/breezmobile/pull/203/commits/0dc52ddfed4b1f897fa04edd555f535068f5e6d0)